src/Authentication/MsalCache.cs

```diff
@@ -0,0 +1,96 @@
+
+namespace Microsoft.Artifacts.Authentication;
+
+public static class MsalCache
+{
+    private static readonly string LocalAppDataLocation = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData, Environment.SpecialFolderOption.Create);
+
+    // from https://github.com/GitCredentialManager/git-credential-manager/blob/df90676d1249759eef8cec57155c27e869503225/src/shared/Microsoft.Git.CredentialManager/Authentication/MicrosoftAuthentication.cs#L277
+    //      The Visual Studio MSAL cache is located at "%LocalAppData%\.IdentityService\msal.cache" on Windows.
+    //      We use the MSAL extension library to provide us consistent cache file access semantics (synchronization, etc)
+    //      as Visual Studio itself follows, as well as other Microsoft developer tools such as the Azure PowerShell CLI.
+    public static string DefaultMsalCacheLocation
+    {
+        get
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // The shared MSAL cache is located at "%LocalAppData%\.IdentityService\msal.cache" on Windows.
+                return Path.Combine(LocalAppDataLocation, ".IdentityService");
+            }
+            else
+            {
+                // The shared MSAL cache metadata is located at "~/.local/.IdentityService/msal.cache" on UNIX.
+                return Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".local", ".IdentityService");
+            }
+        }
+    }
+
+    public static async Task<MsalCacheHelper?> GetMsalCacheHelperAsync(string cacheLocation, ILogger logger)
+    {
+        MsalCacheHelper? helper = null;
+
+        logger.LogTrace($"Using MSAL cache at `{cacheLocation}`");
+
+        const string cacheFileName = "msal.cache";
+
+        // Copied from GCM https://github.com/GitCredentialManager/git-credential-manager/blob/bdc20d91d325d66647f2837ffb4e2b2fe98d7e70/src/shared/Core/Authentication/MicrosoftAuthentication.cs#L371-L407
+        try
+        {
+            var storageProps = CreateTokenCacheProperties(useLinuxFallback: false);
+
+            helper = await MsalCacheHelper.CreateAsync(storageProps);
+
+            helper.VerifyPersistence();
+        }
+        catch (MsalCachePersistenceException ex)
+        {
+            logger.LogWarning("warning: cannot persist Microsoft authentication token cache securely!");
+            logger.LogTrace(ex.ToString());
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                // On macOS sometimes the Keychain returns the "errSecAuthFailed" error - we don't know why
+                // but it appears to be something to do with not being able to access the keychain.
+                // Locking and unlocking (or restarting) often fixes this.
+                logger.LogError(
+                    "warning: there is a problem accessing the login Keychain - either manually lock and unlock the " +
```

Comment:
Do we need to do localization on these logging message ?  Are we doing localization at all currently in the repo?

Reply:
We probably should, but are not currently. The rest of the credential provider (except the MSAL stuff and some ad-hoc places) uses resource files, but we're not building or localizing this at all currently.

Reply:
Went on a journey and ended up re-working how resource generation is done entirely, and added resource files for the new library.